src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,386 @@
+import React, { Component } from "react";
+import { NumberInput, TextArea } from "@carbon/react";
+import { isIpv4CidrOrAddress, transpose } from "lazy-z";
+import PropTypes from "prop-types";
+import { checkNullorEmptyString } from "../../lib";
+import { isIpStringInvalid, isRangeInvalid } from "../../lib/iam-utils";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import { IcseSelect } from "../Dropdowns";
+import { IcseTextInput, IcseNameInput, IcseToggle } from "../Inputs";
+import { SecurityGroupMultiSelect, SubnetMultiSelect } from "../MultiSelects";
+import { IcseFormGroup } from "../Utils";
+import IcseFormTemplate from "../IcseFormTemplate";
+import VpnServerRouteForm from "./VpnServerRouteForm";
+
+class VpnServerForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.data;
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleNumberInputChange = this.handleNumberInputChange.bind(this);
+    this.handleMultiSelectChange = this.handleMultiSelectChange.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.handleAllowedIps = this.handleAllowedIps.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  handleInputChange(event) {
+    let { name, value } = event.target;
+    let stateChangeParams = { [name]: value };
```

Comment:
let's just create a copy of the whole state here using the spread operator (`{...this.state}`) so you can set values directly instead of using transpose